sql/core/src/main/scala/org/apache/spark/sql/catalog/Catalog.scala

```diff
@@ -223,6 +223,9 @@ abstract class Catalog {
    * If this table is cached as an InMemoryRelation, drop the original cached version and make the
    * new version cached lazily.
    *
+   * If the table's schema is inferred at runtime, infer the schema again and update the schema
```

Comment:
@rxin @cloud-fan  I see. Will make a change

FYI, this will change the existing external behavior.